### PR TITLE
Daniil/goldsky management

### PR DESCRIPTION
**Pull request by kekonen**
Package that allows to create duplicate Goldsky environments, in a organised and almost automatic manner.
It produces a bulk of `CREATE OR REPLACE VIEW` SQL statements, which just need to be applied in a SQL console, to create and/or update different Goldsky's nomad environments.

### CODE REVIEW

**Review by xBalbinus — APPROVED** [review]

File: packages/goldsky_management/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+// Dummy super class that leads the dance
+let s = new Setup([prod, stage]);
+
+// Register views. (name, template, required tables/views by name)
+let disp = new ViewTemplate('decoded_dispatch', fs.readFileSync('./views/decodedDispatch.sql', 'utf8'), [`dispatch`]);
+let update = new ViewTemplate('decoded_update', fs.readFileSync('./views/decodedUpdate.sql', 'utf8'), ['update']);
+let events = new ViewTemplate('events', fs.readFileSync('./views/events.sql', 'utf8'), ['decoded_dispatch', 'decoded_update', 'process']);
+let numberMessages = new ViewTemplate('number_messages', fs.readFileSync('./views/numberMessages.sql', 'utf8'), ['events']);
+let recovery = new ViewTemplate('recovery_view', fs.readFileSync('./views/recovery.sql', 'utf8'), ['recovery']);
```
Comment (Imti) [inline packages/goldsky_management/src/index.ts:170]: Can we name this just `recovery`?

File: packages/goldsky_management/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+let processFailure = new ViewTemplate('process_failure_view', fs.readFileSync('./views/processFailure.sql', 'utf8'), ['process_failure']);
```
Comment (Imti) [inline packages/goldsky_management/src/index.ts:171]: Same here, just named `process_failure` to be consistent with the other views.

File: packages/goldsky_management/src/index.ts (hunk 1 above)
Comment (kekonen) [reply packages/goldsky_management/src/index.ts:170]: I wanted to separate the view and the source table. It also simultaneously intersects with the recovery within the package (`...['recovery']);` )

File: packages/goldsky_management/views/decodedDispatch.sql [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  decoded.recipient_address,
+  decoded.message_body,
+  decoded.message__token_body,
+  decoded.message__token__domain,
+  decoded.message__token__id,
+  decoded.action_body,
+  decoded.message__action__type,
+  decoded.message__action__to,
+  decoded.message__action__amount,
+  decoded.message__action__details_hash,
```
Comment (ErinHales) [inline packages/goldsky_management/views/decodedDispatch.sql:151]: Use `decoded.*,`

Hasura makes this more verbose than necessary

File: packages/goldsky_management/views/decodedUpdate.sql [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+update_1.id,
+update_1."timestamp",
+update_1.transaction_hash,
+update_1.contract_id,
+update_1.home_domain,
+update_1.old_root,
+update_1.new_root,
+update_1.signature,
+update_1._gs_chain,
+update_1._gs_gid,
```
Comment (ErinHales) [inline packages/goldsky_management/views/decodedUpdate.sql:15]: Select `update_1.*,`

Hasura makes this more verbose than necessary

File: packages/goldsky_management/views/undispatchedProcess.sql [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  process.id,
+  process."timestamp",
+  process.transaction_hash,
+  process.block_number,
+  process.contract_id,
+  process.message_hash,
+  process.success,
+  process.return_data,
+  process._gs_chain,
+  process._gs_gid
```
Comment (ErinHales) [inline packages/goldsky_management/views/undispatchedProcess.sql:14]: `SELECT * from <%= process; %>`

File: packages/goldsky_management/views/willMintNew.sql [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  events.destination_domain_id,
+  (events.message__token__domain) :: integer AS token_domain,
+  events.message__token__id AS token_id,
+  events.message__action__amount AS amount,
+  events.dispatch_tx,
+  events.relay_tx,
+  events.update_tx,
+  events.process_tx
+FROM
+  <%= events; %>
```
Comment (ErinHales) [inline packages/goldsky_management/views/willMintNew.sql:16]: `SELECT * from <%= events; %>`

File: packages/goldsky_management/views/willMintNewAll.sql [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  events.dispatched_at AS sent_at,
+  events.origin_domain_id,
+  events.destination_domain_id,
+  (events.message__token__domain) :: integer AS token_domain,
+  events.message__token__id AS token_id,
+  events.message__action__amount AS amount,
+  events.dispatch_tx,
+  events.relay_tx,
+  events.update_tx,
+  events.process_tx
```
Comment (ErinHales) [inline packages/goldsky_management/views/willMintNewAll.sql:14]: `SELECT * from <%= events; %>`

File: packages/goldsky_management/views/events.sql [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+          SELECT
+            update_1.old_root,
+            update_1.new_root,
+            update_1.signature,
+            update_1.block,
+            update_1.transaction_hash,
+            update_1."timestamp",
+            update_1.gs_chain_id
+          FROM
+            <%= decoded_update; %> update_1
```
Comment (ErinHales) [inline packages/goldsky_management/views/events.sql:57]: `SELECT * FROM <%= decoded_update %> update_1`

File: packages/goldsky_management/views/events.sql [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+      )
+      LEFT JOIN (
+        SELECT
+          relay_1.transaction_hash,
+          relay_1.block,
+          relay_1."timestamp",
+          relay_1.old_root,
+          relay_1.gs_chain_id
+        FROM
+          <%= decoded_update; %> relay_1
```
Comment (ErinHales) [inline packages/goldsky_management/views/events.sql:80]: `SELECT * FROM <%= decoded_update %> relay_1`

File: packages/goldsky_management/views/bridgeEvents.sql [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        receive_1.transaction_hash,
+        receive_1.block_number,
+        receive_1.contract_id,
+        receive_1.origin_and_nonce,
+        receive_1.token,
+        receive_1.recipient,
+        receive_1.liquidity_provider,
+        receive_1.amount,
+        receive_1._gs_chain,
+        receive_1._gs_gid
```
Comment (ErinHales) [inline packages/goldsky_management/views/bridgeEvents.sql:91]: `SELECT * FROM`

**Review by ErinHales — CHANGES_REQUESTED** [review]

**Pull request merged**